project/ws/app/src/lib/routes/home/components/request-list/create-request-form/create-request-form.component.ts

```diff
@@ -386,7 +386,7 @@ export class CreateRequestFormComponent implements OnInit {
   compAreaSelected(option: any) {
     this.resetCompSubfields()
     this.allCompetencies.forEach((val: any) => {
-      if (option.name === val.name) {
+      if (option.name === val.name && val.children.length) {
```

Comment:
please check for val.children before val.children.length

Reply:
Added the check